README.md

```diff
@@ -365,6 +365,32 @@ Of course, the situation in which a class grows this long is strongly discourage
   }
   ```
 
+- If the only goal is to match on the type of the object, do NOT expand fully all the arguments, as it makes refactoring more difficult and the code more error prone.
```

Comment:
I agree with caution here, especially in a case like this where there are a bunch of back to back integers (you might even call that out.  I think its the same thing as using named arguments).  Maybe this should more be pro/cons of use extractors vs. matching on types:

Pro-extactors:
  - few arguments
  - nested matching

Avoid extractors:
  - Only using some of the arguments
  - Many arguments of the same type